xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp

```diff
@@ -849,6 +855,18 @@ bool CMMALVideo::GetPicture(DVDVideoPicture* pDvdVideoPicture)
     pDvdVideoPicture->iFlags  = DVP_FLAG_ALLOCATED;
     if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_FLAG_USER3)
       pDvdVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_INTERLACED)
+    {
+      pDvdVideoPicture->iFlags |= DVP_FLAG_INTERLACED;
+      if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_TOP_FIELD_FIRST)
+        pDvdVideoPicture->iFlags |= DVP_FLAG_TOP_FIELD_FIRST;
+      if (!firstInterlacedFrame)
+      {
+        pDvdVideoPicture->dts = DVD_NOPTS_VALUE;
+        pDvdVideoPicture->pts = DVD_NOPTS_VALUE;
```

Comment:
Not sure I agree here. With my [1080i60 sample](https://dl.dropboxusercontent.com/u/3669512/temp/1620_20130917060600.mpg) (recording from live TV), I get 30 packets per second from demuxer. 30 packets with timestamps get passed to decoder. I get 30 decoded pictures (frames consisting of two fields) out of decoder.
Any timestamp created for the second deinterlace frames would have to be guessed.